pygfx/helpers/_lights.py

```diff
@@ -213,12 +212,10 @@ def __init__(self, color=None):
             Geometry(positions=positions),
             LineSegmentMaterial(thickness=1.0),
         )
+        self.world.on_update(self._update)
```

Comment:
I didn't realize it in earlier PRs, but the light helpers are apparently attempting to synchronize to changes in their parents by hooking into their matrix world callback. Usually this was called by the renderer just before drawing, but that isn't the case anymore.

If you look at the _update function, it's behavior actually depend on the parent transform. It's just hooking into the render call to synchronize some other properties.

I think this needs to utilize the `before_draw` event instead, just like the Gizmo does.

Reply:
> I think this needs to utilize the before_draw event instead, just like the Gizmo and Controllers do.

The problem with that is that would require these objects to be passed a ref to the renderer somehow :/   Therefore I think in internal system may be more appropriate in this case.

Reply:
Let's pick this one up in a future PR. Exposing `before_draw` will need a larger refactor of the internal event system and I think this PR is massive enough as is.